### DISCUSSION TITLE
Self host using own domain. After delete Alias, unable to re-create the same alias.

### DISCUSSION THREAD

**Discussion by whisperdancer** [simple-login/app#1391] [Self-hosting question]
My self hosted instance is running great now! However, have an issue. My understanding is with own domain I can delete an alias and then create that same alias again. However, when I try to recreate a deleted alias, it won't let me. It seems to be treating my own domain as if it were the simplelogin domain. Is there a workaround to allow me to re-create a deleted alias? It is my own domain so this should be allowed.

Thank you.

**Comment by nguyenkims** [comment]
@whisperdancer SL distinguishes between public domain and custom domain indeed. Public domain can be used by all users and is the one used in the setup. As a workaround, you can remove the deleted alias from the table `deleted_alias` by connecting to the database. Another workaround is to re-add the domain as a custom domain so you can have a total control over it.

**Comment by whisperdancer** [comment]
@nguyenkims When I try to re-add domain as a custom domain, I receive an error message "A custom domain cannot be a built in domain"

**Reply by whisperdancer** [reply]
@nguyenkims I understand you're busy, but would appreciate an answer to this. Thank you!

**Reply by nguyenkims** [reply]
We've pushed the new version `simplelogin/app:4.6.4-beta` that removes that check, can you try to update to this version and retry? The upgrade guide is on https://github.com/simple-login/app/blob/b80e3bb25a4bd27146ae911aa60f58d4b330745d/docs/upgrade.md#L1-L0

**Comment by whisperdancer** [comment]
Thank you @nguyenkims
I upgraded to 4.6.4 beta and am able to add my own domain as a custom domain. I did add the TXT record to verify the domain. However, I noticed it's showing SPF, DKIM, DMARC are not verified. When setting up the self hosted server I added the SPF, DKIM and DMARC for my domain per the setup guide https://github.com/simple-login/app#dns. 

As my public domain is my custom domain ( they are the same domain ) and the DNS records (SPF, DKIM and DMARC) for that domain were added during initial setup, how do you recommend I proceed?

Thanks so much for your help!

**Reply by nguyenkims** [reply]
I just pushed another version 4.6.5-beta that should fix this issue. Can you update your version, remove that domain and retry please?

**Reply by whisperdancer** [reply]
Thanks @nguyenkims I upgraded to 4.6.5. Removed and added back my domain as a custom domain. I've noted the following discrepancies:

It now shows the domain, DKIM and SPF as verified. DMARC still shows unverified.

The MX record shows as verified and the value matches that of the DNS MX record. Same as the value set during initial setup.

For DKIM, it shows as verified and displays this:
```
Record: CNAME
Domain: dkim._domainkey
Value: dkim._domainkey.mydomain.com. 
```
However, this CNAME record doesn't exist in my DNS records for my domain.

For SPF it shows verified and displays:
```
Record: TXT
Domain: mydomain.com or @
Value: v=spf1 include:mydomain.com ~all 
```
This does not match the value of my DNS SPF record.

For DMARC:
Add the following TXT DNS record to your domain.
```
Record: TXT
Domain: _dmarc
Value: v=DMARC1; p=quarantine; pct=100; adkim=s; aspf=s 
```

Your DNS is not correctly set. The TXT record we obtain is:
`v=DMARC1; p=quarantine; adkim=r; aspf=r `

Kindly advise. Thank you.

**Reply by nguyenkims** [reply]
The DKIM and SPF for the default domain can't be set up the same way as a custom domain as otherwise it'll cause a loop in terms of DNS resolution. You can however enable the DMARC if you want, it should already be enabled when you set up SimpleLogin instance.

**Reply by whisperdancer** [reply]
Sorry, I'm still a bit confused. So, I can just add my domain as a custom domain. I don't have to do anything else and it will just work with the existing DNS records setup during initial install, which include DKIM and SPF?

**Reply by nguyenkims** [reply]
Yes that's right. Once this domain is added as a custom domain, you can have full control over it in SL UI, for ex to create aliases without suffix.

**Reply by Slevin90** [reply]
> I just pushed another version 4.6.5-beta that should fix this issue. Can you update your version, remove that domain and retry please?

Good work for the future.
I am however on arm architecture (rapsberry pi) and latest version for it is 3.4.0.

I was able to manually change the files like :4.6.4-beta. And could add my own domain as custom domain. This works.

As whisperdancer I am stuck on the MX verification step. Manually making the changes from 3.4.0 to https://github.com/simple-login/app/commit/6c6add6b284ff79323894dbe440ac82b3df1f2a4  is too big.


Could you help with providing a manual SQL query to work around for now?  Similar like we can manually set Premium users on self-hosted
`docker exec -it sl-db psql -U myuser simplelogin
UPDATE users SET lifetime = TRUE;
exit`


**Update**  Didn't see last weeks dockerhub -beta's were also in arm64. I was able to update to 4.5.6-beta and everything is cool :-)  Keep up the good work!
